v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/SpannerToMySqlCustomTransformationLT.java

```diff
@@ -0,0 +1,155 @@
+
+import com.google.cloud.teleport.metadata.TemplateLoadTest;
+import com.google.cloud.teleport.v2.spanner.migrations.transformation.CustomTransformation;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.common.TestProperties;
+import org.apache.beam.it.gcp.datagenerator.DataGenerator;
+import org.apache.beam.it.jdbc.JDBCResourceManager;
+import org.apache.beam.it.jdbc.MySQLResourceManager;
+import org.apache.beam.it.jdbc.conditions.JDBCRowsCheck;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Category(TemplateLoadTest.class)
+@TemplateLoadTest(SpannerToSourceDb.class)
+@RunWith(JUnit4.class)
+public class SpannerToMySqlCustomTransformationLT extends SpannerToSourceDbLTBase {
+  private static final Logger LOG = LoggerFactory.getLogger(SpannerToMySqlSourceLT.class);
+
+  private String generatorSchemaPath;
+  private final String artifactBucket = TestProperties.artifactBucket();
+  private final String spannerDdlResource = "SpannerToMySqlSourceLT/spanner-schema.sql";
+  private final String sessionFileResource = "SpannerToMySqlCustomTransformationLT/session.json";
+  private final String dataGeneratorSchemaResource =
+      "SpannerToMySqlSourceLT/datagenerator-schema.json";
+  private final String table = "Person";
+  private final int maxWorkers = 50;
+  private final int numWorkers = 20;
+  private PipelineLauncher.LaunchInfo jobInfo;
+  private PipelineLauncher.LaunchInfo readerJobInfo;
+  private final int numShards = 1;
+
+  @Before
+  public void setup() throws IOException, InterruptedException {
+    setupResourceManagers(spannerDdlResource, sessionFileResource, artifactBucket);
+    setupMySQLResourceManager(numShards);
+    generatorSchemaPath =
+        getFullGcsPath(
+            artifactBucket,
+            gcsResourceManager
+                .uploadArtifact(
+                    "input/schema.json",
+                    Resources.getResource(dataGeneratorSchemaResource).getPath())
+                .name());
+
+    createMySQLSchema(jdbcResourceManagers);
+    CustomTransformation customTransformation =
+        CustomTransformation.builder(
+                "input/customShard.jar", "com.custom.CustomTransformationWithShardForLiveIT")
+            .build();
+    createAndUploadJarToGcs(gcsResourceManager);
+    jobInfo = launchDataflowJob(artifactBucket, numWorkers, maxWorkers, customTransformation);
+  }
+
+  @After
+  public void tearDown() {
+    cleanupResourceManagers();
+  }
+
+  @Test
+  public void reverseReplication1KTpsLoadTest()
```

Comment:
Please change the method name to contain context of full test case. `reverseReplication1KTpsWithCustomTransformation` maybe?